pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/PinotKafkaConsumerFactory.java

```diff
@@ -35,4 +35,6 @@ public static PinotKafkaConsumerFactory create(KafkaStreamMetadata kafkaStreamMe
   public abstract PinotKafkaConsumer buildConsumer(@Nonnull String clientId, int partition, KafkaStreamMetadata kafkaStreamMetadata);
 
   public abstract PinotKafkaConsumer buildMetadataFetcher(@Nonnull String clientId, KafkaStreamMetadata kafkaStreamMetadata);
+
+  public abstract KafkaMessageDecoder getDecoder(KafkaLowLevelStreamProviderConfig kafkaStreamProviderConfig) throws Exception;
```

Comment:
Doesnt need to be abstract. You can implement a method here that returns the configured decoder class.